website/src/components/Chat/PluginsChooser.tsx

```diff
@@ -0,0 +1,257 @@
+      // onPluginActivated(newPlugins[index].enabled);
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginEdit = useCallback(
+    (index: number) => {
+      setSelectedPluginIndex(index);
+      onOpen();
+    },
+    [onOpen]
+  );
+
+  const handlePluginRemove = useCallback(
+    (index) => {
+      const updatedPlugins = plugins.filter((_, idx) => idx !== index);
+      setValue("plugins", updatedPlugins, { shouldDirty: true });
+    },
+    [setValue, plugins]
+  );
+
+  const handlePluginSave = useCallback(async () => {
+    if (textareaRef.current) {
+      const newPlugins = [...plugins];
+      newPlugins[selectedPluginIndex!].url = textareaRef.current.value;
+
+      const plugin: PluginEntry = await post(API_ROUTES.GET_PLUGIN_CONFIG, {
+        arg: { plugin: newPlugins[selectedPluginIndex!] },
+      });
+      // disable all existing plugins and enable new one
+      newPlugins.forEach((plugin) => (plugin.enabled = false));
+      newPlugins[selectedPluginIndex!].enabled = true;
+      newPlugins[selectedPluginIndex!].plugin_config = plugin.plugin_config;
+      setValue("plugins", newPlugins, { shouldDirty: true });
+      //onPluginActivated(newPlugins[selectedPluginIndex!].enabled);
+    }
+    onClose();
+  }, [onClose, control, plugins, selectedPluginIndex]);
+
+  const handlePluginAdd = useCallback(() => {
+    const newPlugin: PluginEntry = {
+      url: "",
+      enabled: false,
+    };
+    setValue("plugins", [...plugins, newPlugin]);
+    handlePluginEdit(plugins.length);
+  }, [setValue, plugins, handlePluginEdit]);
+
+  const limitPluginNameLength = (name: string) => {
+    if (!name) {
+      return "";
+    }
+    if (name.length > 15) {
+      return name.substring(0, 15) + "...";
+    }
+    return name;
+  };
+
+  return (
+    <FormControl mb="40px" ref={containerRef}>
+      <Box position="fixed" zIndex="1">
+        <Menu placement="auto">
+          <MenuButton as={Button} pr="3" rightIcon={<Cog />} w={measureDivWidth(containerRef.current) + 15} size="lg">
+            {plugins && plugins[findSelectedPluginIndex()]?.enabled ? (
+              <Box display="flex" justifyContent="flex-start" gap={2}>
+                <Image
+                  src={plugins[findSelectedPluginIndex()]?.plugin_config?.logo_url}
+                  alt="Plugins"
+                  width="25px"
+                  height="25px"
+                  marginRight="5px"
+                />
+                <Text mt="4px" fontSize="15px">
+                  {limitPluginNameLength(plugins[findSelectedPluginIndex()]?.plugin_config?.name_for_human)}
+                </Text>
+              </Box>
+            ) : (
+              <Box display="flex" justifyContent="center" gap={2}>
+                <AttachmentIcon boxSize="5" />
+                {t("plugins")}
+              </Box>
+            )}
+          </MenuButton>
+          <MenuList left="-150px" position="absolute" w="280px">
```

Comment:
please use inset inline start or inset block start to account for ltr and rtl, otherwise you can try with tailwind.

Reply:
I was told, that you will polish ui quirks, and it does not have to be perfect. so I just make it usable in UI.
Do you want me to go trough all and make it perfect now, and ping you after I finish?

Reply:
for L154 you mean something like this maybe: class="inset-x-[-150px] absolute w-[280px]"  ?

Reply:
You are right, we can ignore this now.